### PR TITLE
feat(remote-development): Add distant.nvim

### PR DESCRIPTION
**Pull request by Uzaaft**

### CODE REVIEW

**Review by mehalter — CHANGES_REQUESTED** [review]

File: lua/astrocommunity/remote-development/distant-nvim/init.lua [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  config = function()
+    require("distant").setup {
+      -- Applies Chip's personal settings to every machine you connect to
+
+      -- 1. Ensures that distant servers terminate with no connections
+      -- 2. Provides navigation bindings for remote directories
+      -- 3. Provides keybinding to jump into a remote file's parent directory
+      ["*"] = require("distant.settings").chip_default(),
+    }
+  end,
```
[the remark this answers is not in the document]
Comment (Uzaaft) [reply lua/astrocommunity/remote-development/distant-nvim/init.lua]: Fixed

File: lua/astrocommunity/remote-development/distant-nvim/init.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "DistantInstall",
+    "DistantMetadata",
+    "DistantShell",
+    "DistantShell",
+    "DistantSystemInfo",
+    "DistantClientVersion",
+    "DistantSessionInfo",
+    "DistantCopy",
+  },
+  opts = function(_, opts) require("astronvim.utils").extend_tbl(opts, require("distant.settings").chip_default()) end,
```
Comment (mehalter) [inline lua/astrocommunity/remote-development/distant-nvim/init.lua:17]:
```suggestion
  opts = function(_, opts) opts["*"] = require("distant.settings").chip_default() end,
```

Shouldn't this be this? Based on your previous commit at least

**Review by mehalter — CHANGES_REQUESTED** [review]

File: lua/astrocommunity/remote-development/distant-nvim/init.lua (hunk 2 above)
Comment (Uzaaft) [reply lua/astrocommunity/remote-development/distant-nvim/init.lua:17]: Sorry, oversight

**Review by mehalter — APPROVED** [review]
LGTM!

**Pull request merged**